source/geod24/LocalRest.d

```diff
@@ -303,22 +303,53 @@ public void sleep (Duration timeout)
         dg = If non-null, this delegate will be called when the timer fires
         periodic = Speficies if the timer fires repeatedly or only once
 
+    Returns:
+        A `Timer` instance with the ability to control the timer
+
 *******************************************************************************/
 
-public void setTimer (Duration timeout, void delegate() dg,
+public Timer setTimer (Duration timeout, void delegate() dg,
     bool periodic = false)
 {
     assert(scheduler !is null, "Cannot call this delegate from the main thread");
     assert(dg !is null, "Cannot call this delegate if null");
-    scheduler.schedule(
-    ()
+
+    Timer timer = new Timer(timeout, dg, periodic);
+    scheduler.schedule(() { timer.run(); });
+    return timer;
+}
+
+/// Simple timer
+final class Timer
+{
+    private Duration timeout;
+    private void delegate () dg;
+    // Using a pointer so Timer may be copied and stop() will still work
+    private bool periodic;
+
+    public this (Duration timeout, void delegate() dg, bool periodic)
+    {
+        this.timeout = timeout;
+        this.dg = dg;
+        this.periodic = periodic;
+    }
+
+    // Run a delegate after timeout, and until this.periodic is false
+    private void run ()
     {
         do
         {
             sleep(timeout);
             dg();
-        } while (periodic);
-    });
+        } while (this.periodic);
```

Comment:
This should be:

```
do
{
    sleep(timeout);
    if (this.stopped)
        return;
    dg();
} while (this.periodic);
```

And the `stop` routine should do `this.stopped = false;`, so you will need to introduce a new `bool` variable.

This is because we need to have the ability to **cancel** a timer while it is sleeping.

So this PR needs to introduce two features:
- Stop a reoccurring timer (this.periodic is initially true)
- Cancel a one-shot timer (this.periodic is initially false)

Reply:
I think this is why @Geod24 suggested that this should be could be handled by the Scheduler. Because the scheduler itself could know not to call `run()` if the timer has been cancelled.

That being said, we can improve upon the implementation later. I think the API will remain the same.

Reply:
I suppose so.